### ISSUE TITLE
Your models have changes that are not yet reflected in a migration

### ISSUE THREAD

**Issue by carn1x** [django-oscar/django-oscar#1814]
On Django 1.7, oscar 1.1, after running `manage.py migrate` for the first time, it seems to run fine without error.

Subsequent runs of `managy.py migrate` give the following error:

```
...
Running migrations:
  No migrations to apply.
  Your models have changes that are not yet reflected in a migration, and so won't be applied.
  Run 'manage.py makemigrations' to make new migrations, and then re-run 'manage.py migrate' to apply them.
```

If I run `manage.py makemigrations` the following files are created:

**site-packages/oscar/apps/customer/migrations/0002_auto_20150723_2301.py**:
```
# -*- coding: utf-8 -*-
from __future__ import unicode_literals

from django.db import models, migrations
import oscar.models.fields.autoslugfield
import django.core.validators


class Migration(migrations.Migration):

    dependencies = [
        ('customer', '0001_initial'),
    ]

    operations = [
        migrations.AlterField(
            model_name='communicationeventtype',
            name='code',
            field=oscar.models.fields.autoslugfield.AutoSlugField(populate_from=b'name', validators=[django.core.validators.RegexValidator(regex=b'^[a-zA-Z_][0-9a-zA-Z_]*$', message="Code can only contain the letters a-z, A-Z, digits, and underscores, and can't start with a digit.")], editable=False, max_length=128, separator='_', blank=True, help_text='Code used for looking up this event programmatically', unique=True, verbose_name='Code'),
            preserve_default=True,
        ),
    ]
```

**site-packages/oscar/apps/catalogue/migrations/0006_auto_20150723_2301.py**:
```
# -*- coding: utf-8 -*-
from __future__ import unicode_literals

from django.db import models, migrations
import django.core.validators
import oscar.core.validators


class Migration(migrations.Migration):

    dependencies = [
        ('catalogue', '0005_auto_20150604_1450'),
    ]

    operations = [
        migrations.AlterField(
            model_name='productattribute',
            name='code',
            field=models.SlugField(max_length=128, verbose_name='Code', validators=[django.core.validators.RegexValidator(regex=b'^[a-zA-Z_][0-9a-zA-Z_]*$', message="Code can only contain the letters a-z, A-Z, digits, and underscores, and can't start with a digit"), oscar.core.validators.non_python_keyword]),
            preserve_default=True,
        ),
```

**Comment by carn1x** [comment]
Seems a new issue caused by the Validator change
https://github.com/django-oscar/django-oscar/commit/700242330a6d3fe12159bff276c94b349bfadde6

**Comment by carn1x** [comment]
Looks like the following commits are responsible:

https://github.com/django-oscar/django-oscar/commit/276eba92491acbc99663363115ed491b2342670d
https://github.com/django-oscar/django-oscar/commit/700242330a6d3fe12159bff276c94b349bfadde6

**Comment by slav0nic** [comment]
This migrations break django 1.8/py3 makemigrations:
```./manage.py makemigrations
Traceback (most recent call last):
  File "./manage.py", line 10, in <module>
    execute_from_command_line(sys.argv)
  File "/mnt/data2/work/graymalin/lib/python3.4/site-packages/django/core/management/__init__.py", line 354, in execute_from_command_line
    utility.execute()
  File "/mnt/data2/work/graymalin/lib/python3.4/site-packages/django/core/management/__init__.py", line 346, in execute
    self.fetch_command(subcommand).run_from_argv(self.argv)
  File "/mnt/data2/work/graymalin/lib/python3.4/site-packages/django/core/management/base.py", line 394, in run_from_argv
    self.execute(*args, **cmd_options)
  File "/mnt/data2/work/graymalin/lib/python3.4/site-packages/django/core/management/base.py", line 445, in execute
    output = self.handle(*args, **options)
  File "/mnt/data2/work/graymalin/lib/python3.4/site-packages/django/core/management/commands/makemigrations.py", line 125, in handle
    migration_name=self.migration_name,
  File "/mnt/data2/work/graymalin/lib/python3.4/site-packages/django/db/migrations/autodetector.py", line 43, in changes
    changes = self._detect_changes(convert_apps, graph)
  File "/mnt/data2/work/graymalin/lib/python3.4/site-packages/django/db/migrations/autodetector.py", line 186, in _detect_changes
    self.generate_altered_fields()
  File "/mnt/data2/work/graymalin/lib/python3.4/site-packages/django/db/migrations/autodetector.py", line 850, in generate_altered_fields
    if old_field_dec != new_field_dec:
  File "/mnt/data2/work/graymalin/lib/python3.4/site-packages/django/core/validators.py", line 55, in __eq__
    self.regex.pattern == other.regex.pattern and
AttributeError: 'bytes' object has no attribute 'pattern'
```
problem in regex bytes, without `b` it's work fine
https://botbot.me/freenode/django-oscar/2016-03-03/?msg=61450113&page=1

**Comment by crgwbr** [comment]
@slav0nic I ran across the same issue. Fixed it and submitted pull-request #2030.

**Comment by vkaracic** [comment]
How come that in the ``customer`` migration ``editable=False`` is set?

**Comment by mvantellingen** [comment]
Should be fixed in 1.2.2 / master
